packages/p2p-v2/src/hooks/useSendbird.ts

```diff
@@ -0,0 +1,252 @@
+
+    const {
+        data: sendbirdServiceToken,
+        isError: isErrorSendbirdServiceToken,
+        isSuccess: isSuccessSendbirdServiceToken,
+    } = useSendbirdServiceToken();
+    const { data: advertiserInfo, isSuccess: isSuccessAdvertiserInfo } = useAdvertiserInfo();
+    const { isError: isErrorChatCreate, mutate: createChat } = useChatCreate();
+    const { data: orderInfo, isError: isErrorOrderInfo } = useOrderInfo(orderId);
+    const { data: serverTime, isError: isErrorServerTime } = useServerTime();
+
+    const getUser = async (userId: string, token: string) => {
+        if (sendbirdApiRef?.current) {
+            const user = await sendbirdApiRef.current.connect(userId, token);
+            return user;
+        }
+    };
+
+    const onMessageReceived = useCallback(() => {
+        if (
+            receivedMessage?.channelUrl === chatChannel?.url &&
+            (receivedMessage?.isUserMessage() || receivedMessage?.isFileMessage())
+        ) {
+            setMessages([...messages, createChatMessage(receivedMessage)]);
+        }
+
+        // NOTE: Do not add messages as dependency, this will cause the function to be recreated over and over again
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [chatChannel?.url, receivedMessage]);
+
+    useEffect(() => {
+        onMessageReceived();
+    }, [receivedMessage]);
+
+    const getChannel = async (channelUrl: string) => {
+        if (sendbirdApiRef?.current) {
+            sendbirdApiRef.current.groupChannel.addGroupChannelHandler(
+                'P2P_SENDBIRD_GROUP_CHANNEL_HANDLER',
+                new GroupChannelHandler({
+                    onMessageReceived: (messageReceivedChannel: BaseChannel, _receivedMessage: BaseMessage) =>
+                        setReceivedMessage(_receivedMessage),
+                })
+            );
+            const channel = await sendbirdApiRef.current.groupChannel.getChannel(channelUrl);
+            return channel;
+        }
+    };
+
+    const getMessages = async (channel: GroupChannel, fromTimestamp?: number) => {
+        const messagesFormatted: ChatMessage[] = [];
+        const timestamp = fromTimestamp || serverTime?.server_time_utc || 0;
+
+        const shouldSortFromMostRecent = messages ? messages?.length > 0 : false;
+        const retrievedMessages = await channel.getMessagesByTimestamp(timestamp, {
+            customTypesFilter: [''],
+            isInclusive: false,
+            messageTypeFilter: MessageTypeFilter.ALL,
+            nextResultSize: 0,
+            prevResultSize: 50,
+            reverse: shouldSortFromMostRecent,
+        });
+
+        retrievedMessages.forEach(message => {
+            if (message.isUserMessage() || message.isFileMessage()) {
+                messagesFormatted.push(createChatMessage(message));
+            }
+        });
+        return messagesFormatted;
+    };
+
+    const sendMessage = (message: string) => {
```

Comment:
@adrienne-deriv, are we not going to handle file messages as well?

Reply:
Yeahh I forgotten to add functionality to upload files as well, sendbird store has that function, I'll add another function `sendFile` for it